### PR TITLE
#247 - created BasicAuthSlice

### LINKED ISSUE #247: Creare `BasicAuthSlice`
Let's create `BasicAuthSlice` to john `SliceAuth` and `BasicIdentities` functionality and deprecate `SliceAuth`, `Identities` and `BasicIdentities`.

### PR DESCRIPTION
**Pull request by olenagerasimova**
Closes #247 
I've created `BasicAuthSlice` from `SliceAuth` and `BasicIdentities` and deprecated `Identities`, `SliceAuth` and `BasicIdentities`

### CODE REVIEW

File: src/main/java/com/artipie/http/auth/BasicAuthSlice.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                }
+            ).orElseGet(
+                () -> {
+                    final Response rsp;
+                    if (this.perm.allowed(Permissions.ANY_USER)) {
+                        rsp = this.origin.response(line, headers, body);
+                    } else {
+                        rsp = new RsWithHeaders(
+                            new RsWithStatus(RsStatus.UNAUTHORIZED),
+                            new Headers.From(new WwwAuthenticate("Basic"))
```
Comment (olegmoz) [inline src/main/java/com/artipie/http/auth/BasicAuthSlice.java:101]: @olenagerasimova we have `PREFIX` constant that is also `"Basic"`, but with space. Also `Basic` auth scheme name is used in `Authorization.Basic` class. I'd propose to join them all into single constant.

File: src/test/java/com/artipie/http/auth/BasicAuthSliceTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    @Test
+    void returnsForbiddenIfNotAllowed() {
+        final String name = "john";
+        MatcherAssert.assertThat(
+            new BasicAuthSlice(
+                new SliceSimple(new RsWithStatus(RsStatus.OK)),
+                (user, pswd) -> Optional.of(new Authentication.User(name)),
+                new Permission.ByName("read", (user, action) -> false)
```
Comment (olegmoz) [inline src/test/java/com/artipie/http/auth/BasicAuthSliceTest.java:90]: @olenagerasimova I think we can simplify `new Permission.ByName("read", (user, action) -> false)` to just `(user, action) -> false`

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see two comments

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected. Also, I've added one more test case to make sure headers are parsed correctly.

File: src/main/java/com/artipie/http/auth/BasicAuthSlice.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+/**
+ * Slice with basic authentication.
+ * @since 0.17
+ */
+public final class BasicAuthSlice implements Slice {
+
+    /**
+     * Basic authentication prefix.
+     */
+    public static final String PREFIX = "Basic";
```
Comment (olegmoz) [inline src/main/java/com/artipie/http/auth/BasicAuthSlice.java:50]: @olenagerasimova name of this field is not really appropriate especially when exposed outside of class. It is a "prefix" when you look at header value as a string, but it's meaning is name authentication scheme which could also be `Bearer`, `Digest` etc. I'd suggest to change this field name to `SCHEME`, that would make more sense to one who reads the code.

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, still have one concern, please take a look

**Comment by olenagerasimova** [comment]
@olegmoz thanks, renamed

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/http#248` is not assigned, can't get performer

**Pull request merged**